### ISSUE TITLE
ROOT view/detector offset

### ISSUE THREAD

**Issue by KrisThielemans** [UCL/STIR#938]
The rotation for images corresponding to ROOT files is currently confusing and largely undocumented. There are 2 keywords in the `.hroot` file:
https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/examples/samples/root_header.hroot#L13
https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/examples/samples/root_header.hroot#L45

# offset (num of detectors)
The `offset (num of detectors)` is used to change the detector numbering
https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/src/IO/InputStreamFromROOTFileForCylindricalPET.cxx#L119-L120

The `offset (num detectors)` keyword is defined [here](https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/src/IO/InputStreamFromROOTFile.cxx#L89), but there is no default set for it in [InputStreamFromROOTFile::set_defaults](https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/src/IO/InputStreamFromROOTFile.cxx#L59) which is arguably a bug.

# additional rotations in the code
Unfortunately, going through the code, there are additional offsets being applied
https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/src/IO/InputStreamFromROOTFileForCylindricalPET.cxx#L115-L116
where https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/src/IO/InputStreamFromROOTFileForCylindricalPET.cxx#L53-L55
(note that this uses integer division, which will be surprising) and

https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/src/listmode_buildblock/CListRecordROOT.cxx#L69-L70
where
https://github.com/UCL/STIR/blob/4221fc32585b71ae1d68febba0fef82f1f819a1b/src/listmode_buildblock/CListRecordROOT.cxx#L29
This is a 90 degrees rotation, but it also uses integer division, so this would create some **small unexpected rotation if the total number of crystals is not a multiple of 4.**

# view offset
This is a general STIR functionality (also used in Interfile proj-data etc) that tells the projectors the relation between the image orientation and the projection data (and hence the detectors). If view offset is zero, the first view is assumed to be vertical (and hence first crystal, after the above ROOT-specific detector offsets, assumed to be at the top of the scanner).

# GATE macros
GATE allows arbitrary location of detectors (STIR does not yet). STIR does not read the GATE macros. It is therefore your responsibility to make these 2 match up.

The additional rotations above seem to come from the fact that by default, GATE puts the (middle of the) first module on the horizontal GATE axis (can someone confirm this?). 

As STIR currently ignores block structure (pending #577), the relevant calculation is an approximation, but I believe that if `offset (num of detectors)=0`, we should have roughly
```
view offset (in degrees) = 360 * ( num_blocks_per_bucket*num_crystals_per_block  - 1) / 2. / num_crystals_per_ring
```
However, the `half_block` calculation interferes with this. I thought we had already removed this, but apparently not. @robbietuk what do you think?

# Problems

Aside from the fact that this is all a bit of a mess, changing the crystal numbering with these internal offsets is a bad idea due to potential conflicts with the normalisation code (which will assume that detector 0 starts at a block/bucket to use symmetries).

**Comment by robbietuk** [comment]
An idea for test was discussed earlier and is similar to the a developing TOF test, https://github.com/NikEfth/STIR/pull/23. Simulate a point source in GATE and `generate_image` using STIR. Evaluate each non-scattered and non-random list mode event, the LOR should travel _**near**_ the point source's location and ideally through the voxel. 

Inter-crystal scatter in the GATE system **_may_** present an issue.

**Comment by KrisThielemans** [comment]
`half_block` rotation is removed in #1012. We still need to do the `quarter_of_detectors`. @robbietuk @jvavrek @francescaleek I suggest we do this now as well. This breaks backwards compatibility for ROOT files dramatically, so we should do that only once.

**Comment by robbietuk** [comment]
Agreed. Im working on the test https://github.com/UCL/STIR/pull/991 a bit. However, in its current state, the GATE scanner is rotated 90 degrees, hence `quarter_of_detectors`. We can fix this with modification to the `view_offset (degrees)` later or change the GATE scanner (we probably should do if we can).

**Comment by KrisThielemans** [comment]
Given a bug in ROOT processing #1008, we will fix this in v5.0.1
